Sources/Rendering/Core/Picker/index.js

```diff
@@ -237,134 +389,36 @@ function vtkPicker(publicAPI, model) {
     p1World[3] = 1.0;
     p2World[3] = 1.0;
 
-    if (model.pickFromList) {
-      props = model.pickList;
-    } else {
-      props = renderer.getActors();
-    }
-    const scale = [];
-    props.forEach((prop) => {
-      const mapper = prop.getMapper();
-      pickable = prop.getNestedPickable() && prop.getNestedVisibility();
-
-      if (prop.getProperty().getOpacity?.() <= 0.0) {
-        pickable = false;
-      }
+    const tolerance =
+      computeTolerance(selectionZ, aspect, renderer) * model.tolerance;
 
-      if (pickable) {
-        model.transformMatrix = prop.getMatrix().slice(0);
-        // Webgl need a transpose matrix but we need the untransposed one to project world points
-        // into the right referential
-        mat4.transpose(model.transformMatrix, model.transformMatrix);
-        mat4.invert(model.transformMatrix, model.transformMatrix);
-        // Extract scale
-        const col1 = [
-          model.transformMatrix[0],
-          model.transformMatrix[1],
-          model.transformMatrix[2],
-        ];
-        const col2 = [
-          model.transformMatrix[4],
-          model.transformMatrix[5],
-          model.transformMatrix[6],
-        ];
-        const col3 = [
-          model.transformMatrix[8],
-          model.transformMatrix[9],
-          model.transformMatrix[10],
-        ];
-        scale[0] = vtkMath.norm(col1);
-        scale[1] = vtkMath.norm(col2);
-        scale[2] = vtkMath.norm(col3);
+    pick3DInternal(model.renderer, tolerance, p1World, p2World);
+  };
 
-        vec4.transformMat4(p1Mapper, p1World, model.transformMatrix);
-        vec4.transformMat4(p2Mapper, p2World, model.transformMatrix);
+  publicAPI.pick3DPoint = (selectionPoint, focalPoint, renderer) => {
+    if (!renderer) {
+      throw new Error('renderer cannot be null');
+    }
 
-        p1Mapper[0] /= p1Mapper[3];
-        p1Mapper[1] /= p1Mapper[3];
-        p1Mapper[2] /= p1Mapper[3];
+    initialize();
+    model.renderer = renderer;
 
-        p2Mapper[0] /= p2Mapper[3];
-        p2Mapper[1] /= p2Mapper[3];
-        p2Mapper[2] /= p2Mapper[3];
+    const selectionX = selectionPoint[0];
+    const selectionY = selectionPoint[1];
+    const selectionZ = selectionPoint[2];
 
-        for (let i = 0; i < 3; i++) {
-          ray[i] = p2Mapper[i] - p1Mapper[i];
-        }
+    model.selectionPoint[0] = selectionX;
+    model.selectionPoint[1] = selectionY;
+    model.selectionPoint[2] = selectionZ;
 
-        if (mapper) {
-          bbox.setBounds(mapper.getBounds());
-          bbox.inflate(tolerance);
-        } else {
-          bbox.reset();
-        }
+    const view = renderer.getRenderWindow().getViews()[0];
+    const dims = view.getViewportSize(renderer);
+    const aspect = dims[0] / dims[1];
```

Comment:
won't work if `dims[1]` is 0 (see recent fix by Forrest)

Reply:
see recent commits